services/workflows-service/src/alert/alert.service.ts

```diff
@@ -103,33 +112,136 @@ export class AlertService {
   }
 
   // Function to perform alert checks for each alert definition
-  async checkAllAlerts(): Promise<void> {
+  async checkAllAlerts() {
     const alertDefinitions = await this.getAllAlertDefinitions();
 
     for (const definition of alertDefinitions) {
       const triggered = await this.checkAlert(definition);
+
       if (triggered) {
-        // const ids = await this.createAlert({
-        //   /* ... */
-        // }); // Provide necessary data
+        this.logger.log(`Alert triggered for alert definition ${definition.id}`);
       }
     }
   }
 
   // Specific alert check logic based on the definition
-  private async checkAlert(definition: AlertDefinition): Promise<boolean> {
-    // ...
-    return true;
+  private async checkAlert(alertDefinition: AlertDefinition): Promise<boolean> {
+    let inlineRule: InlineRule;
+
+    const unknownData: unknown = alertDefinition.inlineRule;
+
+    inlineRule = unknownData as InlineRule;
+
+    const ruleExecutionResults = await this.dataAnalyticsService.runInlineRule(
+      alertDefinition.projectId,
+      inlineRule,
+    );
+
+    if (
+      !ruleExecutionResults ||
+      (Array.isArray(ruleExecutionResults) && ruleExecutionResults.length === 0)
+    ) {
+      return false;
+    }
+
+    const alertResponse = {
+      fulfilled: [],
+      rejected: [],
+      skipped: [],
+    } as any;
+
+    const alertsPromises = ruleExecutionResults
+      .filter((aggregatedRow: Record<string, unknown>) => {
+        if (
+          _.some(
+            inlineRule.groupedBy,
+            field => _.isNull(aggregatedRow[field]) || _.isUndefined(aggregatedRow[field]),
+          )
+        ) {
+          this.logger.error(`Alert aggregated row is missing properties `, {
+            inlineRule,
+            aggregatedRow,
+          });
+          return false;
+        }
+        return true;
+      })
+      .map(async (aggregatedRow: object) => {
+        try {
+          const pickedResult = _.pick(aggregatedRow, inlineRule.groupedBy);
+
+          if (await this.isDuplicateAlert(alertDefinition, pickedResult)) {
+            alertResponse.skipped.push({
+              status: AlertExecutionStatus.SKIPPED,
+              alertDefinition,
+              aggregatedRow,
+            });
+          } else {
+            alertResponse.fulfilled.push({
+              status: AlertExecutionStatus.SUCCEEDED,
+              alertDefinition,
+              aggregatedRow,
+              alert: await this.createAlert(alertDefinition, pickedResult),
+            });
+          }
+        } catch (error) {
+          console.error(error);
+
+          alertResponse.rejected.push({
+            status: AlertExecutionStatus.FAILED,
+            error,
+            alertDefinition,
+            aggregatedRow,
+          });
+        }
+      });
+
+    await Promise.all(alertsPromises);
```

Comment:
Does this need to be `allSettled`?

Reply:
nope the promise in handled since we have high level try..catch in the map function